src/cluster.c

```diff
@@ -4207,6 +4214,54 @@ sds clusterGenNodeDescription(clusterNode *node) {
     return ci;
 }
 
+/* Gnerate all nodes slots topology info. Generating one node slots info is
+ * costly in clusterGenNodeDescription(), if there are many nodes in cluster,
+ * it will cost much time and block server to generate all nodes description,
+ * so we can gnerate all firstly and complexity of this function is similar
+ * to generate one node slots info in clusterGenNodeDescription(). */
```

Comment:
I don't understand why we do this up front, you say the complexity is similar to doing it either way. This is just adding another place for this code to be called. It just looks like we're doing extra loops through the cluster nodes?

Reply:
Hi @madolson  They are different. If cluster has 100 nodes, before this commit, we need to separately get every node slot info, Generating total nodes slot info  is O(100 * CLUSTER_SLOTS), but in this commit, we only call this function once, it is O(CLUSTER_SLOTS).

Reply:
Oh, I see now, that's clever. I added some wording suggestions so that intention is clearer.